product_attribute_value_archive/models/product_attribute_value.py

```diff
@@ -18,7 +18,7 @@ def _get_related_variants(self):
         product.
         """
         self.ensure_one()
-        self.flush()
+        self.flush_recordset()
```

Comment:
Actually, here, we must flush all fields used in the query... and not only those of the recordset.

I think we need to do something like this:

```py
self.env["product.product"].flush_model(
    [
        "product_template_attribute_value_ids",
        "product_template_variant_value_ids",
    ]
)
self.env["product.template.attribute.value"].flush_model(["product_attribute_value_id"])
```